federatedscope/core/message.py

```diff
@@ -180,7 +189,11 @@ def build_msg_value(self, value):
         if isinstance(value, list) or isinstance(value, tuple):
             msg_value.list_msg.MergeFrom(self.create_by_type(value))
         elif isinstance(value, dict):
-            msg_value.dict_msg.MergeFrom(self.create_by_type(value))
+            if isinstance(list(value.keys())[0], str):
```

Comment:
What if a `dict` contains string key and int key?

Reply:
Such situation cannot be handled due to the implementation grpc. A workaround can be that we change all the keys into string and save their original types for recovery, which might bring additional cost.

Reply:
IMO, we'd better have a format check with dict values instead of using the first key.